ompi/mca/coll/basic/coll_basic_neighbor_allgather.c

```diff
@@ -143,16 +145,20 @@ mca_coll_basic_neighbor_allgather_graph(const void *sbuf, int scount,
          * a const for the send buffer. */
         rc = MCA_PML_CALL(isend((void *) sbuf, scount, sdtype, edges[neighbor],
                                 MCA_COLL_BASE_TAG_ALLGATHER, MCA_PML_BASE_SEND_STANDARD,
-                                comm, reqs++));
+                                comm, preqs++));
         if (OMPI_SUCCESS != rc) break;
     }
 
     if (OMPI_SUCCESS != rc) {
-        /* should probably try to clean up here */
+        ompi_coll_base_free_reqs( reqs, (2 * neighbor + 1));
```

Comment:
neighbor is what gives the right number of iterations, the +1 is just to provide an upper bound for the < test in ompi_coll_base_free_reqs.